### PR TITLE
Check gcp crs in 

### PR DESCRIPTION
**Pull request by peterm790**
- [x] Closes https://github.com/corteva/rioxarray/issues/646
 - [ ] Tests added
 - [x] Fully documented, including `docs/history.rst` for all changes and `docs/rioxarray.rst` for new API

I have not written any tests related to this change.

### CODE REVIEW

**Comment by snowman2** [comment]
Mind adding a test similar to this one with the CRS = None?
https://github.com/corteva/rioxarray/blob/205052675b2e2d070a699aa55359806e894e39ec/test/integration/test_integration_rioxarray.py#L2924

**Comment by peterm790** [comment]
I have added a test, let me know if this is what you had in mind :)

File: test/integration/test_integration_rioxarray.py [hunk 1]
```diff
@@ -2933,6 +2933,20 @@ def test_rio_write_gcps():
...
 
+def test_rio_write_gcps_no_crs():
+    """
+    Test setting gcps in dataarray, when gcp_crs is not present.
+    """
+    gdal_gcps, _ = _create_gdal_gcps()
+    gcp_crs = None
+
+    darr = xarray.DataArray(1)
+    darr["spatial_ref"] = None
```
Comment (snowman2) [inline test/integration/test_integration_rioxarray.py:2944]: I don't think this line is needed, correct?
```suggestion
```

**Comment by snowman2** [comment]
> I have added a test, let me know if this is what you had in mind

Looks good :+1:. Once the linting issues are resolved, I think this is ready to merge.

File: test/integration/test_integration_rioxarray.py (hunk 1 above)
Comment (peterm790) [reply test/integration/test_integration_rioxarray.py:2944]: I had also thought not but it fails when writing the CRS without it. 

```
KeyError: 'spatial_ref'

During handling of the above exception, another exception occurred:

KeyError                                  Traceback (most recent call last)

      darr = xarray.DataArray(1)
      #darr['spatial_ref'] = None
----> darr.rio.write_gcps(gdal_gcps, None)

File ~/modis/rioxarray/rioxarray/rioxarray.py:1238, in XRasterBase.write_gcps(self, gcps, gcp_crs, grid_mapping_name, inplace)
   1234     data_obj = data_obj.rio.write_crs(
   1235         gcp_crs, grid_mapping_name=grid_mapping_name, inplace=inplace
   1236     )
   1237 geojson_gcps = _convert_gcps_to_geojson(gcps)
-> 1238 data_obj.coords[grid_mapping_name].attrs["gcps"] = geojson_gcps
   1239 return data_obj

File /share/apps/anaconda3-2022.05/envs/pangeo/lib/python3.10/site-packages/xarray/core/coordinates.py:386, in DataArrayCoordinates.__getitem__(self, key)
    385 def __getitem__(self, key: Hashable) -> T_DataArray:
--> 386     return self._data._getitem_coord(key)

File /share/apps/anaconda3-2022.05/envs/pangeo/lib/python3.10/site-packages/xarray/core/dataarray.py:819, in DataArray._getitem_coord(self, key)
    817 except KeyError:
    818     dim_sizes = dict(zip(self.dims, self.shape))
--> 819     _, key, var = _get_virtual_variable(self._coords, key, dim_sizes)
    821 return self._replace_maybe_drop_dims(var, name=key)

File /share/apps/anaconda3-2022.05/envs/pangeo/lib/python3.10/site-packages/xarray/core/dataset.py:186, in _get_virtual_variable(variables, key, dim_sizes)
    184 split_key = key.split(".", 1)
    185 if len(split_key) != 2:
--> 186     raise KeyError(key)
    188 ref_name, var_name = split_key
    189 ref_var = variables[ref_name]

KeyError: 'spatial_ref'
```

Comment (snowman2) [reply test/integration/test_integration_rioxarray.py:2944]: In that case, this won't work how you are expecting. Probably need logic similar to this in `write_gcps`:
https://github.com/corteva/rioxarray/blob/205052675b2e2d070a699aa55359806e894e39ec/rioxarray/rioxarray.py#L588-L634

**Comment by itcarroll** [comment]
This PR just made it possible for me to read MODIS data into an xarray.Dataset. Thank you! Logs are expired on the checks, but I hope this can get patched and merged. You are amazing.

**Comment by snowman2** [comment]
Thanks @peterm790 :+1:
